### PR TITLE
Load ES mappings from static files or dynamically

### PR DESCRIPTION
**Pull request by simonvadee**
- push static mapping files
- improve fhirstore.ElasticSearchEngine to load the ES mappings either from static files or generate them dynamically (from the FHIR spec)
- add the "as_json" feature from fhirpath
- small performance test
- parametrize es_reference_analyzer and es_token_normalizer
- more tests

### CODE REVIEW

File: fhirstore/search_engine.py [hunk 1]
```diff
@@ -28,12 +65,12 @@ def current_url(self):
...
     def get_index_name(self):
         """ """
-        return "fhirstore"
+        return self.es_index
 
     def get_mapping(self, resource_type: str, index_config=None):
         """ """
         try:
-            return {"properties": self.mappings[resource_type]}
+            return self.mappings[resource_type]
```
Comment (Jasopaum) [inline fhirstore/search_engine.py:76]: Did you also change this somewhere else?

File: fhirstore/search_engine.py [hunk 2]
```diff
@@ -1,21 +1,58 @@
...
+
+            except FileNotFoundError as e:
+                raise SearchEngineError(f"Could not find ES mapping file: {e}")
+            except IsADirectoryError:
+                raise SearchEngineError(f"ELASTIC_MAPPING_FILES_DIR should contain only JSON files")
+
+        # load the ES mappings dynamically
+        else:
+            logging.info(
+                "ELASTIC_MAPPING_FILES_DIR should be defined in environment, "
```
Comment (Jasopaum) [inline fhirstore/search_engine.py]: It's not a "should" if we can generate the mappings dynamically, right?

File: fhirstore/search_engine.py [hunk 3]
```diff
@@ -1,21 +1,58 @@
...
+                "generating mappings dynamically"
+            )
+            self.mappings = self.generate_mappings(
```
Comment (Jasopaum) [inline fhirstore/search_engine.py:56]: You also did some changes in fhirpath, right?

File: fhirstore/search_engine.py [hunk 4]
```diff
@@ -1,21 +1,58 @@
...
         )
-        self.mappings = generate_mappings(FHIR_VERSION.R4.name)
+        self.es_index = es_index
+        self.mappings = {}
+
+        # load the ES mappings from static files
+        if ELASTIC_MAPPING_FILES_DIR:
+            logging.info(f"Loading ES index from {ELASTIC_MAPPING_FILES_DIR}...")
+            try:
+                mapping_files = os.listdir(ELASTIC_MAPPING_FILES_DIR)
```
Comment (Jasopaum) [inline fhirstore/search_engine.py]: I think using `glob` would slightly clean the code here.
You can do `glob(ELASTIC_MAPPING_FILES_DIR/**.json)` with this package to get all the files in `ELASTIC_MAPPING_FILES_DIR` (and no need for the `os.path.join`).

File: fhirstore/search_engine.py (hunk 2 above)
Comment (simonvadee) [reply fhirstore/search_engine.py]: right, done

File: fhirstore/search_engine.py (hunk 3 above)
Comment (simonvadee) [reply fhirstore/search_engine.py:56]: I did ! check out https://github.com/nazrulworld/fhirpath/commit/cf0e51845cec8055e6232df2b66af27d93bcf806

File: fhirstore/search_engine.py (hunk 4 above)
Comment (simonvadee) [reply fhirstore/search_engine.py]: cool, thanks done

File: fhirstore/search_engine.py (hunk 1 above)
Comment (simonvadee) [reply fhirstore/search_engine.py:76]: yes, in the generate_mappings method in fhirpath (see nazrulworld/fhirpath@cf0e518)

**Pull request merged**